Sofa/framework/Geometry/src/sofa/geometry/proximity/PointTriangle.h

```diff
@@ -0,0 +1,77 @@
+﻿/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <sofa/geometry/config.h>
+#include <sofa/type/Mat.h>
+#include <sofa/type/Mat_solve_LCP.h>
+
+#include <utility>
+
+namespace sofa::geometry::proximity
+{
+
+template<typename Node,
+     typename T = std::decay_t<decltype(*std::begin(std::declval<Node>()))>,
+     typename = std::enable_if_t<std::is_scalar_v<T>>
+>
+[[nodiscard]]
+constexpr bool computeClosestPointOnTriangleToPoint(
```

Comment:
Why a new file and not adding a method directly in geometry::triangle::computeClosestPointToPoint ?